docs/build/guides/archival/extend-persistent-entry-js.mdx

```diff
@@ -0,0 +1,152 @@
+---
+title: Extend a persistent entry and contract using the JavaScript SDK
+description: How to extend a persistent entry and contract using the JavaScript SDK
+---
+
+Persistent storage provides a durable mechanism to retain data on the network over a long period. User data, contract instances and their corresponding code (wasm hash) are stored as instance storage with persistent durability to ensure that they remain available, and are managed efficiently by the network's archival system.
+
+Each entry made to the persistent storage is stored as a key-value store. An entry consists of:
+
+- A key (identifier for the data)
+- A value (the data itself)
+- A Time-To-Live (TTL) that determines how long the data remains on the network
+
+When the TTL expires, the data is then archived on the network and is no longer accessible unless it is unarchived. To prevent data and contracts from being inaccessible, the TTL can be extended before it expires.
+
+## Extension Process
+
+1. Create the Ledger Footprint
+2. Build a transaction that includes the `extendFootprintTtl` operation with your target TTL and attach the Soroban transaction data (using the footprint).
+3. Prepare the transaction, sign it with your secret key, and submit it to the network.
+
+### Extending a persistent entry
+
+```javascript
+const StellarSdk = require("@stellar/stellar-sdk");
+const { Server } = require("@stellar/stellar-sdk/rpc");
+
+async function extendPersistentEntryTTL(contractId, storageKey, sourceKeypair) {
+  const server = new Server("https://soroban-testnet.stellar.org");
+  const account = await server.getAccount(sourceKeypair.publicKey());
+  const fee = "200100"; // 200_000 covers Soroban overhead + a BASE_FEE of 100 stroops
```

Comment:
Simulation will cover assigning the right fee as part of `prepareTransaction`, so you can just make this the `BASE_FEE` for inclusion into the ledger.